sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/HttpConstants.java

```diff
@@ -321,6 +321,9 @@
 
         // Client generated gateway network error on ReadTimeoutException
         public static final int GATEWAY_ENDPOINT_READ_TIMEOUT = 10002;
+
+        // Client generated request rate too large exception
+        public static final int THROUGHPUT_CONTROL_REQUEST_RATE_TOO_LARGE = 10003;
```

Comment:
please check in the msdata CosmosDB repo whether that substatus code is being used already - would make sense to use a new one (and possibly get a "reservation" in the master repo to avoid conflicts there)

Reply:
Checked here: https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB?path=%2FProduct%2FMicrosoft.Azure.Documents%2FSharedFiles%2FStatusCodes.cs&_a=contents&version=GBmaster

It is safe to use 10003 substatus code.